packages/fxa-settings/src/components/UnitRowTwoStepAuth/index.tsx

```diff
@@ -0,0 +1,126 @@
+  }
+`;
+
+export const UnitRowTwoStepAuth = () => {
+  const { totp } = useAccount();
+  const { exists, verified } = totp;
+  const [modalRevealed, revealModal, hideModal] = useBooleanState();
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const [errorText, setErrorText] = useState<string>();
+  const onError = (e: Error) => {
+    setErrorText(e.message);
+    hideModal();
+    revealAlertBar();
+  };
+  const [disableTwoStepAuth] = useMutation(DESTROY_TOTP_TOKEN_MUTATION, {
+    variables: { input: {} },
+    onCompleted: () => {
+      hideModal();
+      revealAlertBar();
+    },
+    onError,
+    ignoreResults: true,
+    update: (cache) => {
+      cache.modify({
+        fields: {
+          account: (existing) => {
+            return { ...existing, totp: { exists: false, verified: false } };
+          },
+        },
+      });
+    },
+  });
+
+  const conditionalUnitRowProps = exists
+    ? {
+        headerValueClassName: 'text-green-800',
+        headerValue: 'Enabled',
+        ctaText: 'Change',
+        secondaryCtaText: 'Disable',
+        secondaryButtonClassName: 'cta-caution',
+        revealSecondaryModal: revealModal,
+      }
+    : {
+        headerValue: null,
+        noHeaderValueText: 'Not Set',
+        ctaText: 'Add',
+        secondaryCtaText: undefined,
+        revealSecondaryModal: undefined,
+      };
+
+  return (
+    <UnitRow
+      header="Two-step authentication"
+      route="/beta/settings/two_step_authentication"
+      {...conditionalUnitRowProps}
+    >
+      <p className="text-sm mt-3">
+        Prevent someone else from logging in by requiring a unique code only you
+        have access to.
+      </p>
+      {modalRevealed && (
+        <VerifiedSessionGuard onDismiss={hideModal} onError={onError}>
+          <Modal
+            onDismiss={hideModal}
+            onConfirm={disableTwoStepAuth}
+            headerId="two-step-auth-disable-header"
+            descId="two-step-auth-disable-description"
+            confirmText="Disable"
+            confirmBtnClassName="cta-caution"
+          >
+            <h2 className="font-bold text-xl text-center mb-2">
+              Disable Two-step authentication?
+            </h2>
+            {/* "replacing recovery codes" link below will actually drop you into
+            recovery codes flow in the future. */}
+            <p>
+              You won't be able to undo this action. You also have the option of
+              <LinkExternal
+                className="link-blue text-xs mt-2"
+                href="https://support.mozilla.org/en-US/kb/reset-your-firefox-account-password-recovery-keys"
+              >
+                replacing your recovery codes
+              </LinkExternal>
+              .
+            </p>
+          </Modal>
+        </VerifiedSessionGuard>
+      )}
+      {/* TODO: style AlertBar in the error case */}
+      {alertBarRevealed && (
+        <AlertBar onDismiss={hideAlertBar}>
```

Comment:
Oh! `AlertBar` now supports a `type` -- this could be `success`, `error`, or `info`. Could we update this to use the type for when the update was successful or failed?